R/MoBiProject.R

```diff
@@ -0,0 +1,170 @@
+#' @title MoBi Project
+#' @docType class
+#' @description  A MoBi project, containing modules, building blocks, simulations, etc.
+#' @format NULL
+MoBiProject <- R6::R6Class(
+  "MoBiProject",
+  cloneable = FALSE,
+  inherit = ObjectBase,
+  active = list(
+    #' @field sourceFile Path to the file the project was loaded from (read-only)
+    sourceFile = function(value) {
+      private$.readOnlyProperty("sourceFile", value, private$.sourceFile)
+    },
+    #' @field simulationNames Names of the simulations that are present in the project
+    simulationNames = function(value) {
+      if (missing(value)) {
+        model <- self$get("SimulationNames")
+      } else {
+        private$.throwPropertyIsReadonly("simulationNames")
+      }
+    },
+    #' @field parameterIdentificationNames Names of the parameter identifications that are present in the project
+    parameterIdentificationNames = function(value) {
+      if (missing(value)) {
+        model <- self$get("ParameterIdentificationNames")
+      } else {
+        private$.throwPropertyIsReadonly("parameterIdentificationNames")
+      }
+    },
+    #' @field individualsNames Names of the individuals that are present in the project
+    individuals = function(value) {
+      if (missing(value)) {
+        model <- self$get("Individuals")
+      } else {
+        private$.throwPropertyIsReadonly("Individuals")
+      }
+    },
+    #' @field expressionProfilesNames Names of the expression profiles that are present in the project
+    expressionProfiles = function(value) {
+      if (missing(value)) {
+        model <- self$get("ExpressionProfiles")
+      } else {
+        private$.throwPropertyIsReadonly("ExpressionProfiles")
+      }
+    }
+  ),
+  public = list(
+    #' @description
+    #' Initialize a new instance of the class
+    #'
+    #' Should not be directly used. Instead, use function `loadMoBiProject()`
+    #' to load a project.
+    #' @param netObject Reference to `NetObject` .NET MoBi-project object
+    #' @param sourceFile (Optional) File used to load the project
+    #' @returns A new `MoBiProject` object.
+    initialize = function(netObject, sourceFile = NULL) {
+      super$initialize(netObject)
+      private$.sourceFile <- sourceFile
+    },
+
+    #' @description
+    #' Load a simulation from the project
+    #'
+    #' @param simulationName Name of the simulation.
+    #' @returns A `Simulation` object, if the simulation with the given name is present in the
+    #' project. `NULL` if no such simulation is available.
+    getSimulation = function(simulationName) {
```

Comment:
@rwmcintosh @benjaperez1983 this is not supported yet?

Reply:
no, that looks like we have `projectTask.AllSimulations(MoBiProject moBiProject)` but not a single by name.
The API here will need to supply the project from which we retrieve it.
```
getSimulation = function(project, simulationName)
```